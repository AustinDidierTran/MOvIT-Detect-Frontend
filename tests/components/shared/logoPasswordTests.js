/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import LogoPassword from '../../../src/components/shared/logoPassword';

Enzyme.configure({ adapter: new Adapter() });

describe('LogoPassword Tests', () => {
  let wrapper;
  const spy = sinon.spy();
  const props = {
    value: 20,
    iconClass: 'fa fa-id-card',
    placeHolder: 'test',
    onChange: (value) => { spy(value); },
    onKeyPress: (value) => { spy(value); },
  };

  beforeEach(() => {
    spy.resetHistory();
    wrapper = shallow(<LogoPassword {...props} />);
  });

  it('should have proptypes', () => {
    const actualValue = LogoPassword.propTypes;

    const expectedValue = {
      placeHolder: PropTypes.string.isRequired,
      iconClass: PropTypes.string.isRequired,
      value: PropTypes.string,
      onChange: PropTypes.func.isRequired,
      onKeyPress: PropTypes.func.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should call onChange when editing the password field', () => {
    wrapper.find('#password').simulate('change', { target: { value: '20' } });

    expect(spy.calledOnce).toEqual(true);
    expect(spy.getCalls()[0].args[0]).toEqual('20');
  });

  it('should call onKeyPress when simulating a keypress on the password field', () => {
    wrapper.find('#password').simulate('keyPress', 'test');

    expect(spy.calledOnce).toEqual(true);
    expect(spy.getCalls()[0].args[0]).toEqual('test');
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
