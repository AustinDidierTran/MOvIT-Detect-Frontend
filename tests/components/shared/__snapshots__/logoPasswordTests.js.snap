// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoPassword Tests should match the snapshot 1`] = `
<div
  className="form-horizontal row mb-3"
>
  <div
    className="col-1 d-inline-block text-right"
  >
    <span
      style={
        Object {
          "fontSize": "large",
          "paddingTop": "6px",
        }
      }
    >
      <i
        className="fa fa-id-card"
      />
    </span>
  </div>
  <div
    className="col-10 col-lg-8 d-inline-block"
  >
    <input
      className="form-control"
      id="password"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="test"
      type="password"
      value={20}
    />
  </div>
</div>
`;
